rust/operator-binary/src/command.rs

```diff
@@ -0,0 +1,170 @@
+use stackable_operator::commons::s3::S3ConnectionSpec;
+use stackable_operator::commons::tls::{CaCert, Tls, TlsServerVerification, TlsVerification};
+use stackable_trino_crd::authentication::TrinoAuthenticationConfig;
+use stackable_trino_crd::{
+    TrinoCluster, CONFIG_DIR_NAME, DATA_DIR_NAME, ENV_S3_ACCESS_KEY, ENV_S3_SECRET_KEY,
+    HIVE_PROPERTIES, PASSWORD_DB, RW_CONFIG_DIR_NAME, S3_SECRET_DIR_NAME, SECRET_KEY_S3_ACCESS_KEY,
+    SECRET_KEY_S3_SECRET_KEY, STACKABLE_CLIENT_TLS_DIR, STACKABLE_INTERNAL_TLS_DIR,
+    STACKABLE_MOUNT_CLIENT_TLS_DIR, STACKABLE_MOUNT_INTERNAL_TLS_DIR, STACKABLE_TLS_STORE_PASSWORD,
+    SYSTEM_TRUST_STORE, SYSTEM_TRUST_STORE_PASSWORD, USER_PASSWORD_DATA_DIR_NAME,
+};
+
+pub fn container_prepare_args(
+    trino: &TrinoCluster,
+    s3_spec: Option<&S3ConnectionSpec>,
+) -> Vec<String> {
+    let mut args = vec![
+        // Copy system truststore to stackable client tls dir
+        format!("keytool -importkeystore -srckeystore {SYSTEM_TRUST_STORE} -srcstoretype jks -srcstorepass {SYSTEM_TRUST_STORE_PASSWORD} -destkeystore {STACKABLE_CLIENT_TLS_DIR}/truststore.p12 -deststoretype pkcs12 -deststorepass {STACKABLE_TLS_STORE_PASSWORD} -noprompt"),
+    ];
+
+    // User password data
+    if trino.get_authentication().is_some() {
+        args.extend(create_key_and_trust_store(
+            STACKABLE_MOUNT_CLIENT_TLS_DIR,
+            STACKABLE_CLIENT_TLS_DIR,
+            "stackable-client-ca-cert",
+        ));
+        args.extend(chown_and_chmod(USER_PASSWORD_DATA_DIR_NAME));
+    } else if trino.get_client_tls().is_some() {
+        args.extend(create_key_and_trust_store(
+            STACKABLE_MOUNT_CLIENT_TLS_DIR,
+            STACKABLE_CLIENT_TLS_DIR,
+            "stackable-client-ca-cert",
+        ));
+    }
```

Comment:
Why are these cases separate?

Reply:
File based authentication requires a password file to be written (and here we chown and chmod it in line 28)

Reply:
Wouldn't it be more accurate then to do:

```rust
if trino.get_authentication().is_some() {
  // prepare password dir
}
if trino.get_client_tls().is_some() {
  // set up tls stuff
}
```

I get that TLS is required for the authn stuff, but we shouldn't have to spell that out _every_ time something touches anything TLS-related.

Reply:
Now created when tls_enabled (client/authentication) and the user password data only added if auth enabled.